lib/karafka.rb

```diff
@@ -17,6 +17,8 @@
   zeitwerk
 ].each(&method(:require))
 
+require_relative './karafka/railtie' if defined?(Rails)
```

Comment:
Rails starts with `require './config/environment'` => `require './config/application'` => `require 'rails'` as the first thing.
Since `require './config/environment'` should be in `karafka.rb` inside the rails project — `Rails` should be defined at this point.

And it's documented/recommended way (see https://api.rubyonrails.org/classes/Rails/Railtie.html, and I should change `defined?(Rails)` to `defined?(Rails::Railtie)`)

Reply:
Rails starts with bundler running a require on it. I am not talking about the "rails flow" but how bundler loads stuff. It's not always the order you would expect. Hence karafka may be required prior to rails being required. I think that atm bundler requires things in their Gemfile definition order, that is why karafka tries to require rails again. It's not about how Rails start but about the visibility of `Rails` itself.